packages/core/src/app/shipping/Shipping.tsx

```diff
@@ -241,6 +243,7 @@ class Shipping extends Component<ShippingProps & WithCheckoutShippingProps, Ship
             !isEqualAddress(updatedShippingAddress, billingAddress) &&
             !hasRemoteBilling
         ) {
+            promises.push(loadBillingAddressFields())
```

Comment:
We only need to make this API call in shipping step when billing address is same as shipping address and we need to fetch the countries for billing address so that `optional postal code` for some countries is considered and validation is not triggered.